File: 프로그래머스/lv2/12946. 하노이의 탑/하노이의 탑.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    static List<int[]> list = new ArrayList<>();

    public int[][] solution(int n) {
        int[][] answer = {};
        dfs(n, 1, 3, 2);
        answer = list.toArray(new int[list.size()][]);
        return answer;
    }

    static void dfs(int n, int start, int end, int mid) {
        if(n == 1){
            list.add(new int[]{start, end});
            return;
        }
        dfs(n-1, start, mid, end);
        list.add(new int[]{start, end});
        dfs(n-1, mid, end, start);
    }
}  

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
answer = []

def solution(n):
    global answer
    answer = [] 
    dfs(n, 1, 3, 2)
    return answer

def dfs(n, start, end, mid):
    if n == 1:
        answer.append([start, end])
        return
    dfs(n - 1, start, mid, end)
    answer.append([start, end])
    dfs(n - 1, mid, end, start)